tracker_test.go

```diff
@@ -61,3 +63,52 @@ func TestGetTracker(t *testing.T) {
 
 	test.DeleteTrackerOK(t, nil, nil, &controller, result.ID)
 }
+
+// This test ensures that List does not return NIL items.
+// refer : https://github.com/almighty/almighty-core/issues/191
+func TestTrackerListItemsNotNil(t *testing.T) {
+	resource.Require(t, resource.Database)
+	ctx := context.Background()
+	ts := remoteworkitem.NewGormTransactionSupport(db)
+	repo := remoteworkitem.NewTrackerRepository(ts)
+	controller := TrackerController{ts: ts, tRepository: repo, scheduler: rwiScheduler}
+
+	item1, err := repo.Create(ctx, "http://issues.jboss.com", "jira")
+	if err != nil {
+		t.Error("Failed to create tracker item", err)
+	}
+
+	item2, err := repo.Create(ctx, "http://issues.mozilla.com", "github")
+	if err != nil {
+		t.Error("Failed to create tracker item", err)
+	}
+	_, list := test.ListTrackerOK(t, nil, nil, &controller, nil, nil)
+
+	for _, tracker := range list {
+		if tracker == nil {
+			t.Error("Returned Tracker found nil")
+		}
+	}
+	test.DeleteTrackerOK(t, nil, nil, &controller, item1.ID)
+	test.DeleteTrackerOK(t, nil, nil, &controller, item2.ID)
+}
+
+// This test ensures that ID returned by Show is valid.
+// refer : https://github.com/almighty/almighty-core/issues/189
+func TestCreateTrackerValidId(t *testing.T) {
+	resource.Require(t, resource.Database)
+	ctx := context.Background()
+	ts := remoteworkitem.NewGormTransactionSupport(db)
+
+	repo := remoteworkitem.NewTrackerRepository(ts)
+	tracker, err := repo.Create(ctx, "github.com", "github")
+	if err != nil {
+		t.Error("Failed to setup scenario", err)
```

Comment:
Now I am using `test.CreateTrackerCreated` to create object. so these lines are removed.